### DISCUSSION TITLE
Error when reading netCDF file (~200MB in size) which is stored in the azure storage account and mounted as a persistentVolume in AKS using blobfuse

### DISCUSSION THREAD

**Discussion by mozerov** [Azure/azure-storage-fuse#1339] [Q&A] [answered]
Hello,

We are getting the following error when reading netCDF files from the mounted Azure storage container: Failed to create DataSet instance from uri /blobfuse_path_to_volume/file.nc: Error at HDF5 layer.

When the code copies the file from that path to a local /tmp folder then the file can be opened without any issues. Also, the nc file can be read using File.ReadAllBytesAsync("/blobfuse_path_to_volume/file.nc") without any problems.

Do you have any ideas why are we having these issues?

Thanks in advance.

**Comment by bhansalivikas** [comment]
Blobfuse does not care what the file type is or what kind of data it has. If the file was downloaded successfully to local drive then it shall work just fine. Error that you have shared is not coming from blobfuse end. Can you share blobfuse logs to dig more into any error blobfuse might have emitted.

**Reply by mozerov** [reply]
The reason why I think this is somehow related to a combination of blobfuse and the [libnetcdf-dev](https://packages.ubuntu.com/focal/libnetcdf-dev) that is installed in the container where the volume is attached, and the [SDSLite](https://www.nuget.org/packages/SDSLite/3.0.1) is the following code:

```
var blobFuseVolumeFilePath = "/blob/some_file.nc";

File.Copy(blobFuseVolumeFilePath , "/tmp/some_file.nc");

using var dataSet1 = DataSet.Open("/tmp/some_file.nc"); // this line opens the file without any problems
using var dataSet2 = DataSet.Open(blobFuseVolumeFilePath); // this line fails with mentioned exception
```

I haven't found blobfuse-specific logs though. Do you know where can I find them when the container is running in the AKS. We use CSI driver in AKS.

**Reply by mozerov** [reply, marked as answer]
I have finally found the issue, and you were right - it is not caused by the blobfuse. Seems like [NetCDFDataSet](https://github.com/predictionmachines/SDSlite/blob/master/ScientificDataSet/Providers/NetCDF/NetCDFDataSet.cs) needs read/write access to a file but the volume was attached as a readOnly = true.

```
- name: blob-volume
  mountPath: /blob/path
  readOnly: false # It used to be true and it caused the error from the SDSLite
```
